test_infra/test_infra/databases_stack.py

```diff
@@ -0,0 +1,472 @@
+from aws_cdk import aws_iam as iam
+from aws_cdk import aws_kms as kms
+from aws_cdk import aws_rds as rds
+from aws_cdk import aws_redshift as redshift
+from aws_cdk import aws_s3 as s3
+from aws_cdk import aws_secretsmanager as ssm
+from aws_cdk import core as cdk
+
+
+class DatabasesStack(cdk.Stack):  # type: ignore
+    def __init__(
+        self,
+        scope: cdk.Construct,
+        construct_id: str,
+        vpc: ec2.IVpc,
+        bucket: s3.IBucket,
+        key: kms.Key,
+        **kwargs: str,
+    ) -> None:
+        """
+        AWS Data Wrangler Development Databases Infrastructure.
+        Includes Redshift, Aurora PostgreSQL, Aurora MySQL, Microsoft SQL Server.
+        """
+        super().__init__(scope, construct_id, **kwargs)
+
+        self.vpc = vpc
+        self.key = key
+        self.bucket = bucket
+
+        self._set_db_infra()
+        self._set_catalog_encryption()
+        self._setup_redshift()
+        self._setup_postgresql()
+        self._setup_mysql()
+        self._setup_sqlserver()
+
+    def _set_db_infra(self) -> None:
+        self.db_username = "test"
+        self.db_password = cdk.CfnParameter(self, "dbpassword", type="String").value_as_string
+        self.db_password_secret = cdk.SecretValue(self.db_password)
+        self.db_security_group = ec2.SecurityGroup(
+            self,
+            "aws-data-wrangler-database-sg",
+            vpc=self.vpc,
+            description="AWS Data Wrangler Test Arena - Database security group",
+        )
+        self.db_security_group.add_ingress_rule(ec2.Peer.any_ipv4(), ec2.Port.all_traffic())
+        self.rds_subnet_group = rds.SubnetGroup(
+            self,
+            "aws-data-wrangler-rds-subnet-group",
+            description="RDS Database Subnet Group",
+            vpc=self.vpc,
+            vpc_subnets=ec2.SubnetSelection(subnet_type=ec2.SubnetType.PUBLIC),
+        )
+        self.rds_role = iam.Role(
+            self,
+            "aws-data-wrangler-rds-role",
+            assumed_by=iam.ServicePrincipal("rds.amazonaws.com"),
+            inline_policies={
+                "S3": iam.PolicyDocument(
+                    statements=[
+                        iam.PolicyStatement(
+                            effect=iam.Effect.ALLOW,
+                            actions=[
+                                "s3:Get*",
+                                "s3:List*",
+                                "s3:Put*",
+                                "s3:AbortMultipartUpload",
+                            ],
+                            resources=[
+                                self.bucket.bucket_arn,
+                                f"{self.bucket.bucket_arn}/*",
+                            ],
+                        )
+                    ]
+                ),
+            },
+        )
+        cdk.CfnOutput(self, "DatabasesUsername", value=self.db_username)
+        cdk.CfnOutput(self, "DatabasesPassword", value=self.db_password)
```

Comment:
Hmm now that I think about it, we are just storing the password in the output section anyways... Perhaps we should move away from that and use an SSM parameter that we can from within our conftest.py?

Reply:
Yeah, at this point I only ported templates we had to CDK but I agree there are many things to improve. Like, passwords in cleartext make me sweat, too, even if it's just test infra.